modules/firebase/firebase.go

```diff
@@ -0,0 +1,127 @@
+package firebase
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"reflect"
+	"slices"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+// Container represents the Firebase container type used in the module
+type Container struct {
+	testcontainers.Container
+}
+
+const rootFilePath = "/srv/firebase"
+
+// ErrRootNotProvided is returned when the root path is not provided
+var ErrRootNotProvided = errors.New("firebase root not provided (WithRoot is required)")
+
+func gatherPorts(config partialFirebaseConfig) ([]string, error) {
+	var ports []string
+
+	v := reflect.ValueOf(config.Emulators)
+	for i := 0; i < v.NumField(); i++ {
+		emulator := v.Field(i)
+		if emulator.Kind() != reflect.Struct {
+			continue
+		}
+		name := v.Type().Field(i).Name
+
+		enabledF := emulator.FieldByName("Enabled")
+		if enabledF != (reflect.Value{}) && !enabledF.Bool() {
+			continue
+		}
+
+		hostF := emulator.FieldByName("Host")
+		portF := emulator.FieldByName("Port")
+		websocketPortF := emulator.FieldByName("WebsocketPort")
+
+		if hostF != (reflect.Value{}) && !hostF.IsZero() {
+			host := hostF.String()
+			if host != "0.0.0.0" {
+				return nil, fmt.Errorf("config specified %s emulator host on non public ip: %s", name, host)
+			}
```

Comment:
question: this seems to limit host to 0.0.0.0, if so why have the field? that said I'm not sure this check is needed, do you have an example of why it is?